mrf/era-1/basic-syntax-#9.js

```diff
@@ -0,0 +1,189 @@
+// add number a to b
+function add(a, b) {
+  return a + b;
+}
+
+//! test add func
+console.assert(add(1, 1) === 2, { a: 1, b: 1, function: add.name });
+
+// flip a coin and log the result (head or tail)
+function flipACoin() {
+  let result = Math.random();
+
+  if (result >= 0.5) return "HEAD";
+  else return "TAIL";
+}
+
+//! test flipACoin func
+console.log("flipACoin:", flipACoin());
+
+// concat two strings
+function concat(stringA, stringB) {
+  // return stringA + stringB;
+  return stringA.concat(stringB);
+}
+
+//! Test concat func
+console.log("concat: ", concat("abc", "def"));
+
+// 2^x = ?
+function power2(x) {
+  // return Math.pow(2, x);
+  return 2 ** x;
+}
+
+//! test power2 func
+console.log("power2 of 10:", power2(10));
+
+// user object with first name and last name, both capitalized
+// e.g. getUser('hello', 'world') is { firstName: 'Hello', lastName: 'World' }
+function getUser(firstName, lastName) {
+  firstName = firstName.charAt(0).toUpperCase() + firstName.slice(1);
+  lastName = lastName.charAt(0).toUpperCase() + lastName.slice(1);
+  const arr = { firstName: firstName, lastName: lastName };
+  return arr;
+}
+
+//! test getUser function
+console.log("getUser whami:", getUser("mmd", "mmdi"));
+
+// greet someone!
+// e.g. greet('ConnecMent') is "Hello, ConnecMent! Hope you the best."
+function greet(name) {
+  // return "Hello, " + name + " Hope you the best.";
+  return `Hello, ${name} Hope you the best.`;
+}
+
+//! test greet func
+console.log("Greet:", greet("Mr MRF"));
+
+// get array of n random numbers
+function getNRandomNumbers(n) {
+  const final = new Array(n).fill();
+
+  // for (let i = 0; i < n; i++) {
+  //   final[i] = Math.random();
+  // }
+  // return final;
+
+  // or
+  return final.map((x) => (x = Math.random()));
```

Comment:
You don't need ‍`x` at all. This can be re-written as 
```suggestion
  return final.map(() => Math.random());
  // or even
  return final.map(Math.random);
```

Reply:
got it